db/sqldb/migrations/0003_domain_types.up.sql

```diff
@@ -0,0 +1,242 @@
+BEGIN;
+
+DROP TABLE user_account;
+
+DROP TYPE auth_provider;
+
+CREATE TYPE language AS ENUM (
+    'en',
+    'de',
+    'fr',
+    'es'
+);
+
+CREATE TYPE authn_mechanism AS ENUM (
+    'EMAIL_AND_PASS'
+);
+
+CREATE TABLE pacta_user (
+    id TEXT PRIMARY KEY,
+    authn_mechanism authn_mechanism NOT NULL,
+    authn_id TEXT NOT NULL,
+    -- What the user actually entered during sign up.
+    entered_email TEXT NOT NULL UNIQUE,
+    -- The validated, cleaned, consistently cased email address - excludes plus aliases, for example.
+    canonical_email TEXT NOT NULL UNIQUE,
```

Comment:
What do we plan on doing with this? Just because emails are an absolute minefield for "consistent parsing", see GMail's "optional periods" as an example.

Reply:
My only goal here was to provide a backstop to prevent plus-aliasing/case-shifting to generate multiple accounts. Keeping record at the DB layer of some canonical representation prevents dupes. Keeping record of the original at the DB layer is important to use the right email in things like emails (where we want to respect plus aliases, ex).

Reply:
The only problem with this is that the user will still have created two Azure accounts, which could be confusing. I'm sure Azure supports some variant of this, we should probably just move this check up to that layer if possible.